### PR TITLE
Use rubocop config from voxpupuli-rubocop

### PR DESCRIPTION
**Pull request by bastelfreak**
Also contains https://github.com/voxpupuli/beaker_puppet_helpers/pull/4

### CODE REVIEW

**Review by ekohl — CHANGES_REQUESTED** [review]
Some cops have been fixed and https://github.com/voxpupuli/beaker_puppet_helpers/pull/13 will fix more.

File: beaker_puppet_helpers.gemspec [hunk 1]
```diff
@@ -15,7 +15,13 @@ Gem::Specification.new do |s|
   s.files         = `git ls-files`.split("\n")
   s.require_paths = ['lib']
 
+  # Testing dependencies
```
Comment (bastelfreak) [inline beaker_puppet_helpers.gemspec:18]: @ekohl should we add those 4 gems as runtime deps to beaker? Then we don't need to list them in each beaker plugin.

**Comment by ekohl** [comment]
@bastelfreak should we update this to https://github.com/voxpupuli/voxpupuli-rubocop?

**Comment by bastelfreak** [comment]
yes,will do

File: .rubocop_todo.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+# Offense count: 28
+RSpec/StubbedMock:
+  Exclude:
+    - 'spec/beaker_puppet_helpers/dsl_spec.rb'
+    - 'spec/beaker_puppet_helpers/install_utils_spec.rb'
+    - 'spec/beaker_puppet_helpers/module_utils_spec.rb'
+
+# Offense count: 44
+RSpec/SubjectStub:
```
Comment (ekohl) [inline .rubocop_todo.yml:42]: I don't know how else to test the DSL so I think this should just be disabled.

File: .rubocop_todo.yml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+RSpec/MultipleExpectations:
+  Max: 7
+
+# Offense count: 4
+# Configuration parameters: AllowedGroups.
+RSpec/NestedGroups:
+  Max: 4
+
+# Offense count: 4
+RSpec/RepeatedDescription:
```
Comment (ekohl) [inline .rubocop_todo.yml:30]: Legitimate problem: https://github.com/voxpupuli/beaker_puppet_helpers/pull/18

File: Rakefile [hunk 4]
```diff
@@ -3,9 +3,14 @@
 begin
   require 'rubocop/rake_task'
 rescue LoadError
-  # No Rubocop
+  # RuboCop is an optional group
 else
-  RuboCop::RakeTask.new
+  RuboCop::RakeTask.new(:rubocop) do |task|
```
Comment (ekohl) [inline Rakefile:8]: Given the repetition, should we have some wrapper in voxpupuli-rubocop?

**Review by ekohl — COMMENTED** [review]
I'm kind of leaning to disabling the cops that are in TODO.

File: Rakefile (hunk 4 above)
Comment (bastelfreak) [reply Rakefile:8]: we have, but in a other PR you didnt like :D voxpupuli-rubocop defines the rake task and we could include that.

Comment (bastelfreak) [reply Rakefile:8]: I added https://github.com/voxpupuli/voxpupuli-rubocop/blob/master/lib/voxpupuli/rubocop/rake.rb . I'm also happy for other ideas but I also think that's something that could be improved in the future.

File: .rubocop_todo.yml (hunk 2 above)
Comment (bastelfreak) [reply .rubocop_todo.yml]: disabled it

File: .rubocop_todo.yml [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+RSpec/MultipleExpectations:
+  Max: 7
+
+# Offense count: 4
+# Configuration parameters: AllowedGroups.
+RSpec/NestedGroups:
+  Max: 4
+
+# Offense count: 28
+RSpec/StubbedMock:
```
Comment (ekohl) [inline .rubocop_todo.yml:30]: I think we should also disable this.

File: beaker_puppet_helpers.gemspec [hunk 6]
```diff
@@ -15,6 +15,9 @@ Gem::Specification.new do |s|
...
+  s.add_development_dependency 'voxpupuli-rubocop', '~> 1.2'
```
Comment (ekohl) [inline beaker_puppet_helpers.gemspec:19]: Can we declare this in the Gemfile where it was previously defined?

File: Rakefile (hunk 4 above)
Comment (ekohl) [reply Rakefile:8]: Let's just use that then.

File: beaker_puppet_helpers.gemspec (hunk 6 above)
Comment (bastelfreak) [reply beaker_puppet_helpers.gemspec]: done

File: .rubocop_todo.yml (hunk 5 above)
Comment (bastelfreak) [reply .rubocop_todo.yml]: done

File: Rakefile (hunk 4 above)
Comment (bastelfreak) [reply Rakefile]: done

File: Gemfile [hunk 7]
```diff
@@ -9,19 +9,14 @@ gem 'rake', '~> 13.0', groups: %i[development test release]
 group :development do
   gem 'rdoc'
   gem 'redcarpet'
+  gem 'voxpupuli-rubocop', '~> 1.2'
```
Comment (ekohl) [inline Gemfile:12]: Any reason not to have a rubocop group anymore?

File: Rakefile [hunk 8]
```diff
@@ -1,12 +1,6 @@
 # frozen_string_literal: true
 
-begin
-  require 'rubocop/rake_task'
-rescue LoadError
-  # No Rubocop
-else
-  RuboCop::RakeTask.new
-end
+require 'voxpupuli/rubocop/rake'
```
Comment (ekohl) [inline Rakefile:3]: Can you add a begin/rescue around it so `BUNDLE_WITHOUT=development` still allows you to run the tests?

File: Gemfile (hunk 7 above)
Comment (bastelfreak) [reply Gemfile]: moved it to its own group

File: Rakefile (hunk 8 above)
Comment (bastelfreak) [reply Rakefile]: done

**Pull request merged**